src/main/java/com/blamejared/crafttweaker/impl/loot/conditions/CTLootConditionBuilder.java

```diff
@@ -0,0 +1,139 @@
+package com.blamejared.crafttweaker.impl.loot.conditions;
+
+import com.blamejared.crafttweaker.api.CraftTweakerAPI;
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker.api.data.IData;
+import com.blamejared.crafttweaker.api.loot.conditions.ILootCondition;
+import com.blamejared.crafttweaker.impl.loot.conditions.crafttweaker.JsonLootConditionTypeBuilder;
+import com.blamejared.crafttweaker.impl.loot.conditions.vanilla.AlternativeLootConditionTypeBuilder;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import net.minecraft.util.ResourceLocation;
+import org.openzen.zencode.java.ZenCodeType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+@ZenRegister
+@ZenCodeType.Name("crafttweaker.api.loot.conditions.LootConditionBuilder")
+@Document("vanilla/api/loot/conditions/LootConditionBuilder")
+public final class CTLootConditionBuilder {
+    private static final Map<Class<? extends ILootConditionTypeBuilder>, Function<CTLootConditionBuilder, ? extends ILootConditionTypeBuilder>> BUILDERS = new HashMap<>();
+
+    private final List<ILootCondition> conditions;
+
+    private CTLootConditionBuilder() {
+        this.conditions = new ArrayList<>();
+    }
+
+    // Registration
+    public static <T extends ILootConditionTypeBuilder> void register(final Class<T> typeToken, final Function<CTLootConditionBuilder, T> creator) {
+        if (BUILDERS.containsKey(typeToken)) {
+            throw new IllegalStateException("A builder for the given type '" + typeToken.getName() + "' was already registered");
+        }
+        BUILDERS.put(typeToken, creator);
+        CraftTweakerAPI.logDebug("Successfully registered loot condition type builder for '%s' as '%s'", typeToken, creator);
+    }
+
+    public static <T extends ILootConditionTypeBuilder> void register(final Class<T> typeToken, final Supplier<T> creator) {
+        register(typeToken, ignore -> creator.get());
+    }
+
+    // Creation
+    @ZenCodeType.Method
+    public static CTLootConditionBuilder create() {
+        return new CTLootConditionBuilder();
+    }
+
+    @ZenCodeType.Method
+    public static CTLootConditionBuilder createInAnd() {
+        return create();
+    }
```

Comment:
There actually isn't a reason for it. I just added for consistency with `createInOr` in case someone prefers making the base condition the builder is created with more explicit. I can definitely remove it if you feel it's completely redundant.